sentiment_analysis/scripts/train.py

```diff
@@ -0,0 +1,136 @@
+import json
+import time
+from pathlib import Path
+from typing import Optional
+
+import numpy as np
+import tensorflow as tf
+from dotenv import load_dotenv
+from pydantic import BaseSettings
+from sklearn.model_selection import train_test_split
+from transformers import DistilBertTokenizer, TFDistilBertForSequenceClassification
+
+GPU_AVAILABLE = tf.test.is_gpu_available()
+
+load_dotenv(".env")  # need to run script from root of project to read this
```

Comment:
should this `.env` file be committed? I dont see it, does it have vars that I will need to run the code?

Reply:
I think generally not as it contains paths specific to my system

Reply:
OK I think the paths for storing data and models should be generic i.e data should be in say `./data/` folder and models, encoders etc in `./artifacts/` or similar, where it is assumed these are relative paths from the project root. We should aim for zero user specific configuration i.e no absolute paths specific to someones machine

Reply:
Also the training code should check for these folders and create them if they don't exist (out of scope for this PR though)

Reply:
I am making a sample folder and env file atm, and working on them being absolute yes

Reply:
though personally I prefer using pre commit for this task, if you want to run the commit hooks without actually commit you can run 
```
pre-commit run --all-files
```
the reason I prefer this is that then we have a single file (precommit config) which defines what we are using for code styling/linting and a single place to call it from.

Reply:
Also if you want to run something specifically like black you can do 
```
pre-commit run black  --all-files
```